src/crunch.rs

```diff
@@ -0,0 +1,63 @@
+pub(crate) mod crnlib;
+pub(crate) mod crn_utils;
+pub(crate) mod crn_consts;
+pub(crate) mod crn_static_huffman_data_model;
+pub(crate) mod crn_symbol_codec;
+pub(crate) mod crn_unpacker;
+pub mod crn_decomp;
+use crate::bcn;
+extern crate alloc;
+
+struct CrunchDecodeHandler{
+    format: crnlib::crn_format,
+    dxt_data: alloc::vec::Vec<u8>
+}
+
+fn crunch_unpack_level<'vec>(data: &[u8], data_size: u32, level_index: u32) -> Result<CrunchDecodeHandler, &'static str> {
+    let mut tex_info: crn_decomp::crn_texture_info = crn_decomp::crn_texture_info::default();
+    if tex_info.crnd_get_texture_info(data, data_size) == false {
+        return Err("Invalid crunch texture encoding.");
+    }
+    let mut p_context: crn_unpacker::crn_unpacker<'_> = match crn_decomp::crnd_unpack_begin(&data, data_size){
+        Ok(p_context) => p_context,
+        Err(res) => return Err(res)
+    };
+    let width = core::cmp::max(1, tex_info.m_width >> level_index);
```

Comment:
Importing max via `use core::cmp::max` would make this easier to read.

Reply:
or just `1.max(tex_info.m_width >> level_index)`